### DISCUSSION TITLE
No JVM shared library file (jvm.dll) found. Try setting up the JAVA_HOME environment variable properly.

### DISCUSSION THREAD

**Discussion by Kevin5487** [pwollstadt/IDTxl#63] [Q&A]
Hi,I am a student doing project about multivariate-TE. And this is my first time using this package.
After the installation I tried the multivariateTE example and this error ran out.
I search for help but still got no idea for this.
It would be a huge favor if someone can fix this problem.

Adding data with properties: 5 processes, 1000 samples, 5 replications
overwriting existing data
JVMNotFoundException Traceback (most recent call last)
in
16
17 # c) Run analysis
---> 18 results = network_analysis.analyse_network(settings=settings, data=data)
19
20 # d) Plot inferred network to console and via matplotlib

c:\users\yaolab\idtxl\idtxl\multivariate_te.py in analyse_network(self, settings, data, targets, sources)
156 print('\n####### analysing target with index {0} from list {1}'
157 .format(t, targets))
--> 158 res_single = self.analyse_single_target(
159 settings, data, targets[t], sources[t])
160 results.combine_results(res_single)

c:\users\yaolab\idtxl\idtxl\multivariate_te.py in analyse_single_target(self, settings, data, target, sources)
268 """
269 # Check input and clean up object if it was used before.
--> 270 self._initialise(settings, data, sources, target)
271
272 # Main algorithm.

c:\users\yaolab\idtxl\idtxl\network_inference.py in _initialise(self, settings, data, sources, target)
369
370 # Set CMI estimator.
--> 371 self._set_cmi_estimator()
372
373 # Check the provided target and sources.

c:\users\yaolab\idtxl\idtxl\network_analysis.py in _set_cmi_estimator(self)
207 self._cmi_estimator_local = EstimatorClass(self.settings)
208 else:
--> 209 self._cmi_estimator = EstimatorClass(self.settings)
210
211 def _separate_realisations(self, idx_full, idx_single):

c:\users\yaolab\idtxl\idtxl\estimators_jidt.py in init(self, settings)
1323 settings = self._check_settings(settings)
1324 # Start JAVA virtual machine and create JAVA object.
-> 1325 self._start_jvm()
1326 CalcClass = (jp.JPackage('infodynamics.measures.continuous.gaussian').
1327 ConditionalMutualInfoCalculatorMultiVariateGaussian)

c:\users\yaolab\idtxl\idtxl\estimators_jidt.py in _start_jvm(self)
59 jar_location = resource_filename(name, 'infodynamics.jar')
60 if not jp.isJVMStarted():
---> 61 jp.startJVM(jp.getDefaultJVMPath(), '-ea', ('-Djava.class.path=' +
62 jar_location))
63

~\Anaconda3\envs\idtxl\lib\site-packages\jpype_jvmfinder.py in getDefaultJVMPath()
72 else:
73 finder = LinuxJVMFinder()
---> 74 return finder.get_jvm_path()
75
76

~\Anaconda3\envs\idtxl\lib\site-packages\jpype_jvmfinder.py in get_jvm_path(self)
210 if jvm_notsupport_ext is not None:
211 raise jvm_notsupport_ext
--> 212 raise JVMNotFoundException("No JVM shared library file ({0}) "
213 "found. Try setting up the JAVA_HOME "
214 "environment variable properly."

JVMNotFoundException: No JVM shared library file (jvm.dll) found. Try setting up the JAVA_HOME environment variable properly.

**Comment by jlizier** [comment]
I think the suggestion in the error message from jpype "Try setting up the JAVA_HOME environment variable properly" is key.

Try googling how to do that in your environment, try it, and if it doesn't work then send more info. (As @mwibral said in #61 i.e. OS version, and the java virtual machine (jvm) your are using?)

**Comment by RaabeM** [comment]
For me `conda install bioconda::java-jdk` worked (Ubuntu 20.04).
